### ISSUE TITLE
(Important) RAM Usage Still Too High (8GB+)

### ISSUE THREAD

**Issue by jadenquinn** [mezz/JustEnoughItems#1893]
I have 32GB of RAM that is as fast as I will need for the next 15 years. I cannot afford 64 right now. JEI uses at least 7GB of RAM by itself just from loading the game, not even in a world yet. The main problem here that I cannot seem to fix without rebuilding the modpack again (Which I have done at least 12 times already, even that mostly doesn't fix squat) is unbearable Lag-Spikes-Of-Death upon 5 seconds after a world finishes loading in. I know that this is a memory usage issue since Task Manager tells me that Minecraft is maxing out it's 19 freaking GB of RAM I allocate to it! it actually goes a few GB above said allocation limit, but ya know. I am not going to rebuild the modpack again, but I might have to not use JEI as a whole anymore. This is a very important thing to fix, it's not acceptible anymore. Please don't keep on ignoring the worst modded RAM issue that currently exists. (I have built around 30 Fully-Working Modpacks, so I know what I am doing) I know this since around half of those modpacks are in Minecraft Version 1.12.2. I always keep my mods up-to-date, unless some mods that hook into each-other are only working on epicific versions of each mod involved.
Thank you for reading and thinking about this,
-jaden-

**Comment by jadenquinn** [comment]
Here is my spec sheet: 
[REDHOTBOX.txt](https://github.com/mezz/JustEnoughItems/files/4315210/REDHOTBOX.txt)

**Comment by jadenquinn** [comment]
No point in postponing this any more, eventualls someone is going to make a fork of your mod and fix things that you won't. I will not take "It's your problem", or "No, it doesn't have a RAM usage issue" for answers. Actually do something about it please.

**Comment by TehNut** [comment]
You seem to be forgetting that devs have lives. We have jobs that take precedent over modding. If we don't have time in our life to work on mods, we won't. JEI is open source and anybody is able to fork, investigate the problem, and submit a PR. As you can guess, nobody has taken the time to do that, or even show interest in doing it. 

Your latest comment here was very rude and highly unnecessary. I don't remember mezz ever throwing the blame on users (without a valid reason, at least).

Neither you, nor the author other issue you commented in have provided a mod list. For all anybody knows, it could be a completely different mod causing JEI to eat up memory.

**Comment by TeamDman** [comment]
Allocating MC too much ram can cause issues in general.
Also would be helpful to provide any JVM args being used.

**Comment by StargateMC** [comment]
This is an issue.... though @jadenquinn certainly isn't contributing to resolving it in a positive way.

JEI effectively doubles memory usage on our 1.12 pack, mainly impacting on people with weaker CPUs (even if they have the same 8Gb memory allocation).

It causes regular spikes in memory usage and the client to lock up in its entirety on occasion. I believe this is GC not being able to keep up.

I've tried every combination of mods I can think of, including reducing the pack from 80 mods to 30... but removing JEI is the only solution that is workable for us.  

If a pack has more than say 10 mods it starts going downhill performance wise.

**Comment by LiskoSlayer63** [comment]
My friends have this issue too but I don't. I've allocated 8GB RAM and it loads 150+ mods under 2mins. BUT my friends have identical modpack and 8GB causes OutOfMemory. After allocating more memory they're able to load the game, but it takes 15-30min.


One of my friend did NOT have this issue, but then he **reinstalled Minecraft and this issue occurs everytime after that**. He also sent me this:

```
[Client thread/INFO]: Starting JEI took 3.650 min
[Client thread/INFO]: Building ingredient filter took 3.610 min
[Client thread/INFO]: Starting JEI took 3.650 min
```

So there is something wrong with JEI, but we don't know what causes it.

**Comment by LiskoSlayer63** [comment]
As some people were suggesting, the slow startup time is definetily caused by VSync when Optifine is installed.

The reason why it started after reinstall is because the VSync is ON by default. I wrote a very simple code to check if VSync is enabled at PreInitalization phase and turn it off. Then it turns it on after the main menu has opened. This reduced the startup time from 27min to under 2min!

I also noticed, that I do not have any memory issues anymore, but I don't know if it's related or not.
